### PR TITLE
Adding collision sets to CollisionResult

### PR DESCRIPTION
**Pull request by mxgrey**
This pull request addresses the discussion in PR #668

### CODE REVIEW

File: dart/collision/Contact.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
  * All rights reserved.
  *
  * Author(s): Jeongseok Lee <jslee02@gmail.com>
+ *            Michael X. Grey <mxgrey@gatech.edu>
```
Comment (jslee02) [inline dart/collision/Contact.cpp]: Nitpick: We've added a comma at the end of the line when adding authors.

**Comment by jslee02** [comment]
Overall, looks good. Could we add a simple test for this?

**Comment by mxgrey** [comment]
Yep, I'm working on a test right now.

File: dart/collision/Result.cpp [hunk 2]
```diff
@@ -74,15 +81,73 @@ const std::vector<Contact>& CollisionResult::getContacts() const
 }
 
 //==============================================================================
+const std::unordered_set<const dynamics::BodyNode*>
```
Comment (jslee02) [inline dart/collision/Result.cpp]: Maybe returning const reference instead of just const?

File: dart/collision/Result.cpp [hunk 3]
```diff
@@ -74,15 +81,73 @@ const std::vector<Contact>& CollisionResult::getContacts() const
...
+    return;
+  }
+
+  const dynamics::ShapeFrame* frame = object->getShapeFrame();
+  mCollidingShapeFrames.insert(frame);
+
+  if(frame->isShapeNode())
+  {
+    const dynamics::ShapeNode* node =
+        dynamic_cast<const dynamics::ShapeNode*>(frame);
```
Comment (jslee02) [inline dart/collision/Result.cpp]: We could use `asShapeNode()` instead of dynamic casting.

Comment (mxgrey) [inline dart/collision/Result.cpp]: I considered that. I'm not actually sure which would be cheaper: dynamic_cast or the v-table lookup for a virtual function. The difference is probably negligible, so I'm fine with either.

Comment (jslee02) [inline dart/collision/Result.cpp]: Yeah, actually I also wondered that when adding `asShapeNode()`. But I found some post like [this](http://stackoverflow.com/a/9959645) saying virtual function is cheaper so added `asShapeNode()` for these use cases.

Comment (jslee02) [inline dart/collision/Result.cpp]: Maybe we would like to add comment encouraging `asShapeNode()` rather than dynamic_cast when possible.

**Pull request merged**